code/modules/projectiles/gun.dm

```diff
@@ -274,7 +300,40 @@
 	else
 		return
 
+/obj/item/weapon/gun/proc/force_attach(obj/item/weapon/gun_attachment/AT)
+	if(istype(AT))
+		AT.on_attach(src)
+		AT.forceMove(src)
+		attachments += AT
+
 /obj/item/weapon/gun/attackby(obj/item/I, mob/user, params)
+	if(istype(I, /obj/item/weapon/screwdriver))
+		if(customizable)
+			for(var/obj/item/weapon/gun_attachment/A in attachments)
+				user << "<span class='notice'>You unscrew [A] from [src].</span>"
+				A.on_remove(src)
+				A.forceMove(get_turf(src))
+				attachments -= A
+
+	if(istype(I, /obj/item/weapon/gun_attachment))
+		var/obj/item/weapon/gun_attachment/AT = I
+		if(customizable)
+			if(attachments.len >= 6)
+				user << "[src] already has too many attachments!"
+				return
+			for(var/obj/item/weapon/gun_attachment/A in attachments)
+				if(istype(AT, A.type))
```

Comment:
I'm not doing this BTW, this sounds like too much effort and I shouldn't have to add a new slot and recode how attachments work when a single if check works just fine.